server/src/main/java/org/apache/lucene/search/uhighlight/CustomUnifiedHighlighter.java

```diff
@@ -113,7 +113,8 @@ public CustomUnifiedHighlighter(
         int noMatchSize,
         int maxPassages,
         Predicate<String> fieldMatcher,
-        int maxAnalyzedOffset
+        int maxAnalyzedOffset,
```

Comment:
Question: do we need to keep track of both `maxAnalyzedOffset` && `fieldMaxAnalyzedOffset`? I think we could use `Math.min(maxAnalyzedOffset, fieldMaxAnalyzedOffset)`, no?

Reply:
Actually with this implementation it could only be lower, I don't know if the maxAnalyzedOffset has an impact when indexing as it is an index option. Because of this if fieldMaxAnalyzedOffset is higher than maxAnalyzedOffset and the field lenght is higher than maxAnalyzedOffset an exception still occurs.

Reply:
Than we could go with `min(fieldMaxAnalyzedOffset, maxAnalyzedOffset)`, right?

Reply:
Right but that will change the current behavior, because now if the text is bigger than the maxAnalyzedOffset an exception is throw, with the suggested approach it will pass, but will not highlight items after maxAnalyzedOffset. For me this behavior is better, but it will break some tests.

Reply:
Yes, this can be done in the constructor, and I think that your if statement is exactly what I intend. I will update this.

Reply:
Actually  it should be 
`if ((offsetSource == OffsetSource.ANALYSIS)
            && (fieldValueLength > maxAnalyzedOffset && fieldMaxAnalyzedIsNotValid)) {` as it must only throw the exception if the `fieldMaxAnalyzedIsNotValid`. This is because when it is valid the text will be cut fit `maxAnalyzedOffset`.

Reply:
We can cast another exception for when the   `fieldMaxAnalyzedOffset > maxAnalyzedOffset`, I think that this should be better

Reply:
Please see if now it is more clear what is going on.